File: go/structs/Stack.go
package structs

import "fmt"

// Stack stack structure
type Stack struct {
	data []int
}

// Push push an item to the stack
func (s *Stack) Push(item int) {
	if len(s.data) == 0 {
		// s.data = make([]int, 0, 10)
		s.data[0] = item
	} else {
		for i := 1; i < len(s.data)-1; i++ {
			s.data[i+1] = s.data[i]
		}
		s.data[0] = item
	}
}

// Pop pop the top item from the stack
func (s *Stack) Pop() int {
	popped := s.data[0]
	s.data = s.data[1:]
	return popped
}

// Print print the stack
func (s Stack) Print() {
	for i := 0; i < len(s.data); i++ {
		fmt.Printf("%d ", s.data[i])
	}
}

File: python/structs/Stack.py
class Stack:
    def __init__(self):
        self.stack = []
        self.size = 0

    def push(self, e):
        self.size += 1
        self.stack.append(9)
        for i in range(self.size - 1, 0, -1):
            self.stack[i] = self.stack[i - 1]
        self.stack[0] = e

    def pop(self):
        temp = self.stack[0]
        self.stack.pop(0)
        self.size -= 1
        return temp
        # for i in range(self.size - 1):
        #     self.stack[i] = self.stack[i + 1]
        # del self.stack[self.size - 1]
        # self.size -= 1

    def __str__(self):
        return self.stack.__str__()

    def __len__(self):
        return len(self.stack)


def test():
    s = Stack()
    s.push(1)
    s.push(2)
    s.push(3)
    print(f"Length: {len(s)}\nValues: {s}\n")
    print(f"Popped: {s.pop()}\n")
    print(f"Length: {len(s)}\nValues: {s}\n")

test()